### PR TITLE
fix(ProtoBuf): `Component.evidence` optional

### LINKED ISSUE #422: `component.evidence` is repeated in proto and object in jsonschema
Looks like this got missed somehow

I think it must be `optional` in the proto as well.

https://github.com/CycloneDX/specification/blob/master/schema/bom-1.6.proto#L136

https://github.com/CycloneDX/specification/blob/master/schema/bom-1.6.schema.json#L2079

Originally reported [here](https://github.com/CycloneDX/specification/issues/272#issuecomment-1868576165)

### PR DESCRIPTION
**Pull request by jkowalleck**
fixes #422 
by reverting the unreleased https://github.com/CycloneDX/specification/commit/19a153072690dfb2e8475ea3fa2e09a657d0cef6  & https://github.com/CycloneDX/specification/commit/acc5f3a003f26a6f165d83e5f4f7706546ccc055
as discussed here: https://github.com/CycloneDX/specification/issues/422#issuecomment-2454961082

### CODE REVIEW

**Comment by jkowalleck** [comment]
@nscuro @andreas-hilti ,

May I ask you for a review? 

Would you consider this a breaking change? I've read https://protobuf.dev/programming-guides/dos-donts/#repeated-to-scalarin that subject matter.

**Review by andreas-hilti — COMMENTED** [review]
For protobufs with a single evidence entry, it would be compatible, compare also:
https://protobuf.dev/programming-guides/proto3/#updating
However, for protobufs with multiple evidence entries, the merging of the elements will lead to a different behavior than what is specified in the comment (namely ignoring all but the first entry).
Thus, I tend to think it would be a breaking change.

**Comment by jkowalleck** [comment]
@nscuro @andreas-hilti ,

May I ask you for a review?
The actual 1.6.0 diff would be the following: https://github.com/CycloneDX/specification/compare/1.6...jkowalleck:fork_CycloneDX-specification:fix/protobuuf-component-evidence-optional#diff-31a634760e9b4432c392ead00601567422a8cc12ac462dead1a7f7ab9fa90fdb

**Review by nscuro — APPROVED** [review]

File: schema/bom-1.6.proto [hunk 1]
```diff
@@ -133,8 +133,8 @@ message Component {
   repeated Component components = 21;
   // Specifies optional, custom, properties
   repeated Property properties = 22;
-  // Specifies optional license and copyright evidence. Only the first item in the optional repeated list is to be taken into account; every other item in the list is to be ignored/omitted.
```
Comment (jkowalleck) [inline schema/bom-1.6.proto:136]: > Only the first item in the optional repeated list is to be taken into account; every other item in the list is to be ignored/omitted.

this was an early idea to solve the same issue. it was never publishe / released.
therefore, it is not binding.

**Pull request merged**